pallets/transaction-multi-payment/src/lib.rs

```diff
@@ -110,13 +110,16 @@ decl_module! {
 		/// Set currency in which transaction fees are paid.
 		/// This is feeless transaction.
 		/// Selected currency must have non-zero balance otherwise is not allowed to be set.
-		#[weight = (<T as Config>::WeightInfo::set_currency(), Pays::No)]
+		// REVIEW: You might want to only make the transaction free if it goes through successfully.
+		// Makes it harder to DoS the chain.
+		#[weight = (<T as Config>::WeightInfo::set_currency())]
```

Comment:
we want this to be usable for user with 0 balance in default currency, so the currency it actually has can be set. we need a different way to throttle this.

Reply:
Yeah, you will probably want to find some way to throttle it.

Reply:
We might want to subtract small amount of the target currency on success to discourage infinite free flipping between the currencies. Not sure if we should calculate the spot price from the pool though as this will bring another dependency on an Oracle. Fail should not get in the block if I'm right.